pytorch_lightning/core/lightning.py

```diff
@@ -1715,50 +1715,12 @@ def unfreeze(self) -> None:
         self.train()
 
     def get_progress_bar_dict(self) -> Dict[str, Union[int, str]]:
-        r"""
-        Implement this to override the default items displayed in the progress bar.
-        By default it includes the average loss value, split index of BPTT (if used)
-        and the version of the experiment when using a logger.
-
-        .. code-block::
-
-            Epoch 1:   4%|▎         | 40/1095 [00:03<01:37, 10.84it/s, loss=4.501, v_num=10]
-
-        Here is an example how to override the defaults:
-
-        .. code-block:: python
-
-            def get_progress_bar_dict(self):
-                # don't show the version number
-                items = super().get_progress_bar_dict()
-                items.pop("v_num", None)
-                return items
-
-        Return:
-            Dictionary with the items to be displayed in the progress bar.
-        """
-        # call .item() only once but store elements without graphs
-        running_train_loss = self.trainer.fit_loop.running_loss.mean()
-        avg_training_loss = None
-        if running_train_loss is not None:
-            avg_training_loss = running_train_loss.cpu().item()
-        elif self.automatic_optimization:
-            avg_training_loss = float("NaN")
-
-        tqdm_dict = {}
-        if avg_training_loss is not None:
-            tqdm_dict["loss"] = f"{avg_training_loss:.3g}"
-
-        if self.truncated_bptt_steps > 0:
-            tqdm_dict["split_idx"] = self.trainer.fit_loop.split_idx
-
-        if self.trainer.logger is not None and self.trainer.logger.version is not None:
-            version = self.trainer.logger.version
-            # show last 4 places of long version strings
-            version = version[-4:] if isinstance(version, str) else version
-            tqdm_dict["v_num"] = version
-
-        return tqdm_dict
+        rank_zero_deprecation(
+            "The `LightningModule.get_progress_bar_dict` method was deprecated in v1.5 and will be removed in v1.7."
+            " Please use the `ProgressBar.get_progress_bar_dict` callback instead.",
+            stacklevel=5,
+        )
+        return self.trainer.progress_bar_callback.get_progress_bar_dict(self)
```

Comment:
this will call the deprecation warning multiple times, which will be very noisy. 
- let's add the deprecation to the docstring
- we can potentially check in the trainer if this method is overridden, and issue the deprecation warning then. either we could reuse this, or add a similar one for deprecation that emits warnings instead of exceptions: https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/trainer/configuration_validator.py

Reply:
sg will do!

Reply:
Note that the deprecations are warnings and should be printed once per process